test cases/unit/99 install_plan/meson.build

```diff
@@ -0,0 +1,75 @@
+project('install_plan', 'c')
```

Comment:
Why have you copied the whole install_tag directory? You could just reuse the same directory in a new unit test.

Reply:
I added more things and removed unnecessary ones. I *generally* think it's better to keep these kinds of things apart, as any change would impact the other tests as well. Although they are similar, `install_plan` was based on `install_tag`, they are not the same. If you think it we should merge them, I can do that though.

Reply:
Using a single test directory where possible is a performance optimization that helps avoid the CI growing so time consuming that it times out or takes even longer to  complete than it already does -- each additional unit test means doing another full subprocess to `meson setup && ninja` and neither forking nor project configuration is exactly instant.

So if things are similar enough to fold into another test then that's generally good to do, but if it's needed, it's needed.

It's a pretty subjective measurement so not what I'd call a real blocker.

Reply:
Well, in this case it's a manual unit test, so even if they share the folder it would still run twice.

I'm not sure what you added/removed compared to the install_tag test, but the goal was to cover all installation code paths. For example if we later add a new function in meson install_something() and we add that in the test for install_tag, I want the CI to fail and reminds me that I also have to test it for install_plan, no?

Reply:
OTOH, it also adds some dependency between the two tests, adding or modifying stuff in the test project will likely affect the other tests and require some adjusting. IMO, things diverge enough to justify having different test projects, but like you said, it's subjective.

Reply:
> install_something() and we add that in the test for install_tag, I want the CI to fail and reminds me that I also have to test it for install_plan, no?

Not necessarily, I think, as it is not always needed, but we can merge the projects. I had initially used the same project, but then added a new one because I needed more stuff than the tag tests did, and did not need some stuff the tag tests did, so :shrug:
If you want, I will have a look at this tomorrow if I have time.

Reply:
Maybe it's just personal taste, but I hate copy/pasted code with subtle changes here and there, it means if we want to change something we have many places where we need to do the same change. Now I'm wondering what you added/removed, maybe they make sense to test for tags too?

Anyway, this is just a small nitpicking after all.

Reply:
I renamed `98 install tag` to `98 install all targets` and merged the changes I add in `99 install_plan`.